docs/spec/v1beta2/receivers.md

```diff
@@ -0,0 +1,453 @@
+   ```
+
+2. Apply the resource on the cluster:
+
+   ```sh
+   kubectl -n flux-system apply -f github-receiver.yaml
+   ```
+
+3. Run `kubectl -n flux-system get receivers` to see the generated URL:
+
+   ```console
+   NAME              READY   STATUS                                                                        
+   github-receiver   True    Receiver initialised for path: /hook/bed6d00b5555b1603e1f59b94d7fdbca58089cb5663633fb83f2815dc626d92b
+   ```
+
+4. On GitHub, navigate to your repository and click on the "Add webhook" button under "Settings/Webhooks".
+   Fill the form with:
+   - **Payload URL**: compose the address using the receiver ingress hostname and the generated path `https://<hostname>/<webhookPath>`.
+   - **Secret**: use the token string
+
+## Writing a Receiver spec
+
+As with all other Kubernetes config, a Receiver needs `apiVersion`,
+`kind`, and `metadata` fields. The name of a Receiver object must be a
+valid [DNS subdomain name](https://kubernetes.io/docs/concepts/overview/working-with-objects/names#dns-subdomain-names).
+A Receiver also needs a
+[`.spec` section](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#spec-and-status).
+
+### Type
+
+`.spec.type` is a required field that specifies which SaaS API to use.
+
+The supported receiver types are:
+
+| Receiver                                            | Type           |
+|-----------------------------------------------------|----------------|
+| [Generic webhook](#generic-receiver)                | `generic`      |
+| [Generic webhook with HMAC](#generic-hmac-receiver) | `generic-hmac` |
+| [GitHub](#github-receiver)                          | `github`       |
+| [Gitea](#github-receiver)                           | `github`       |
+| [GitLab](#gitlab-receiver)                          | `gitlab`       |
+| [Bitbucket server](#bitbucket-server-receiver)      | `bitbucket`    |
+| [Harbor](#harbor-receiver)                          | `harbor`       |
+| [DockerHub](#dockerhub-receiver)                    | `dockerhub`    |
+| [Quay](#quay-receiver)                              | `quay`         |
+| [Nexus](#nexus-receiver)                            | `nexus`        |
+| [Azure Container Registry](#acr-receiver)           | `acr`          |
+| [Google Container Registry](#gcr-receiver)          | `gcr`          |
+
+### Events filtering
+
+`.spec.events` in an optional field to specify a list of event types
+that this Receiver should handle. If left empty, all events are handled.
+
+### Resources
+
+`.spec.resources` is a required field to specify which Flux Custom Resources
+should be reconciled when an event is received.
+
+A resource entry must contain the following fields:
+- `apiVersion` is the Flux Custom Resource API group and version such as `source.toolkit.fluxcd.io/v1beta2`.
+- `kind` is the Flux Custom Resource `.kind` such as GitRepository, OCIRepository, HelmRepository, etc.
```

Comment:
Similar to Provider, as it's a spec doc, maybe we should list all the supported values for `kind`.